### PR TITLE
Melodic devel gbp refine action (test PR)

### PR DESCRIPTION
**Pull request by jcnorby**

### CODE REVIEW

File: global_body_planner/include/global_body_planner/planning_utils.h [hunk 1]
```diff
@@ -72,6 +70,36 @@ struct PlannerConfig {
...
+
+  void loadVectors() {
+    // Load the gravity vector
+    G_VEC << 0, 0, -G_CONST;
+
+    // Load the reachability test points
+    reachability_points_body << 0.5 * ROBOT_L, -0.5 * ROBOT_L, 0.5 * ROBOT_L,
+        -0.5 * ROBOT_L, 0.5 * ROBOT_W, 0.5 * ROBOT_W, -0.5 * ROBOT_W,
+        -0.5 * ROBOT_W, 0, 0, 0, 0;
+    // Load the collision test points
```
Comment (ardalantj) [inline global_body_planner/include/global_body_planner/planning_utils.h:97]: space between the lines

File: global_body_planner/include/global_body_planner/planning_utils.h [hunk 2]
```diff
@@ -234,6 +262,9 @@ void getInterpPlan(const FullState &start_state,
 double getZFromState(const State &s, const PlannerConfig &planner_config);
 double getPitchFromState(const State &s, const PlannerConfig &planner_config);
 double getDzFromState(const State &s, const PlannerConfig &planner_config);
+double getZClearance(const Eigen::Vector3d pos,
```
Comment (ardalantj) [inline global_body_planner/include/global_body_planner/planning_utils.h:265]: const &

File: global_body_planner/src/global_body_planner.cpp [hunk 3]
```diff
@@ -104,7 +104,7 @@ GlobalBodyPlanner::GlobalBodyPlanner(ros::NodeHandle nh) {
                            planner_config_.GOAL_BOUNDS);
   quad_utils::loadROSParam(nh, "global_body_planner/max_planning_time",
                            planner_config_.MAX_TIME);
-  planner_config_.G_VEC << 0, 0, -planner_config_.G_CONST;
+  planner_config_.loadVectors();
```
Comment (ardalantj) [inline global_body_planner/src/global_body_planner.cpp:107]: loadParamsIntoEigen?

File: global_body_planner/test/test_kinematics.cpp [hunk 4]
```diff
@@ -78,11 +78,13 @@ TEST(GlobalBodyPlannerTest, testLeapAction) {
 
     if (!s.isApprox(result.s_new)) {
       std::cout << "Reverse:" << std::endl;
+      std::cout << "is_valid_reverse = " << ((is_valid_reverse) ? 1 : 0) << std::endl;
       printState(s);
       printState(result.s_new);
       printState(s_init_1);
       printState(s_init_2);
       printAction(a);
+      return;
```
Comment (ardalantj) [inline global_body_planner/test/test_kinematics.cpp:87]: add explicit error handling

File: global_body_planner/test/test_performance.cpp [hunk 5]
```diff
@@ -13,7 +13,7 @@ TEST(GlobalBodyPlannerTest, testIsValidStateTime) {
   // Create planner and configuration
   PlannerClass P(FORWARD);
   PlannerConfig planner_config;
-  planner_config.G_VEC << 0, 0, -planner_config.G_CONST;
+  planner_config.loadVectors();
```
Comment (ardalantj) [inline global_body_planner/test/test_performance.cpp:16]: Test fixture for all the tests?

File: local_planner/include/local_planner/local_planner.h [hunk 6]
```diff
@@ -88,6 +88,12 @@ class LocalPlanner {
    */
   void getStateAndTwistInput();
 
+  /**
+   * @brief Function identify indices which violate full order constraints
+   * @return Vector of indices which violate full order constraints
+   */
+  Eigen::VectorXi getInvalidRegions();
```
Comment (ardalantj) [inline local_planner/include/local_planner/local_planner.h:95]: Return ref or input as ref

File: local_planner/src/local_footstep_planner.cpp [hunk 7]
```diff
@@ -102,6 +102,18 @@ void LocalFootstepPlanner::computeContactSchedule(
       contact_schedule[i] = nominal_contact_schedule_[(i + phase) % period_];
     }
   }
+  // Check the primitive plan to see if there's standing or flight phase
+  for (size_t i = 0; i < horizon_length_; i++) {
+    // Leaping and landing
+    if (ref_primitive_plan(i) == 1 || ref_primitive_plan(i) == 3) {
```
Comment (ardalantj) [inline local_planner/src/local_footstep_planner.cpp:108]: Either define variables as const or add comments to clarify

File: local_planner/src/local_footstep_planner.cpp [hunk 8]
```diff
@@ -102,6 +102,18 @@ void LocalFootstepPlanner::computeContactSchedule(
...
+      std::fill(contact_schedule.at(i).begin(), contact_schedule.at(i).end(),
+                true);
+    } else if (ref_primitive_plan(i) == 2) {
```
Comment (ardalantj) [inline local_planner/src/local_footstep_planner.cpp:111]: Same as above

File: local_planner/src/local_planner.cpp [hunk 9]
```diff
@@ -511,20 +575,35 @@ bool LocalPlanner::computeLocalPlan() {
...
+  Eigen::VectorXi complexity_schedule(N_ + 1);
+  complexity_schedule.fill(0);
+
   // Compute grf position considering the toe radius
-  Eigen::MatrixXd grf_positions_body = foot_positions_body_;
+  Eigen::MatrixXd grf_positions = foot_positions_world_;
   for (size_t i = 0; i < 4; i++) {
-    grf_positions_body.col(3 * i + 2) =
-        foot_positions_body_.col(3 * i + 2).array() - toe_radius;
+    // grf_positions.col(3 * i + 2) = foot_positions_body_.col(3 * i +
```
Comment (ardalantj) [inline local_planner/src/local_planner.cpp:584]: commented code?

File: local_planner/src/local_planner.cpp [hunk 10]
```diff
@@ -511,20 +575,35 @@ bool LocalPlanner::computeLocalPlan() {
...
-          current_state_, ref_body_plan_, grf_positions_body, contact_schedule_,
-          ref_ground_height_, first_element_duration_, same_plan_index_,
-          body_plan_, grf_plan_))
+          current_full_state, ref_body_plan_, grf_positions,
+          foot_velocities_world_, contact_schedule_, ref_ground_height_,
+          first_element_duration_, same_plan_index_, ref_primitive_plan_,
+          complexity_schedule, body_plan_, grf_plan_))
     return false;
 
+  // Eigen::VectorXi complexity_schedule_adaptive = getInvalidRegions();
```
Comment (ardalantj) [inline local_planner/src/local_planner.cpp:605]: commented code?

File: nmpc_controller/src/quad_nlp_utils.cpp [hunk 11]
```diff
@@ -0,0 +1,323 @@
...
+  iRow_mat_.resize(num_sys_id_);
+  jCol_mat_.resize(num_sys_id_);
+
+  // Iterate through all functions and compute metadata
+  for (int sys_id = 0; sys_id < num_sys_id_; sys_id++) {
+    // Resize vector data
+    iRow_mat_[sys_id].resize(num_func_id_);
+    jCol_mat_[sys_id].resize(num_func_id_);
+
+    for (int func_id = 0; func_id < num_func_id_; func_id++) {
```
Comment (ardalantj) [inline nmpc_controller/src/quad_nlp_utils.cpp:216]: add commnets

**Comment by jcnorby** [comment]
(This was to demo code reviews, not a real PR)

**Pull request closed without merging**